### PR TITLE
require ip version 5.1.0

### LINKED ISSUE #189: require NCEPLIBS-ip-5.1.0 in the cmake build (after it is released)
Soon @AlexanderRichert-NOAA will release NCEPLIBS-ip-5.1.0 which will have the rotated lat/lon interpolation.

Once that happens, does wgrib2 always require NCEPLIBS-ip-5.1.0 or better? Or can we still work with older versions of ip/sp? 

In cmake we have:

```
# If user wants to use NCEPLIBS-ip, find it and the sp library.
message(STATUS "Checking if the user want to use NCEPLIBS-ip...")
if(USE_IPOLATES)
  find_package(ip CONFIG REQUIRED)
  list(APPEND definitions_list -DIPOLATES_LIB="ipolates_lib_4")
  list(APPEND definitions_list -DUSE_IPOLATES)
endif()
```

We need to decide about the version, whether sp is ever required, and check that in the cmake.

I think this assumes ip 5.0.0.

In fact, for

### PR DESCRIPTION
**Pull request by AlysonStahl-NOAA**
Fixes #189

### CODE REVIEW

File: spack/package.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def url_for_version(self, version):
     )
     variant(
         "ipolates",
-        default="3",
+        default="0",
         description="Use to interpolate to new grids (0 = OFF, 1 = ip, 3 = ip2)",
         values=("0", "1", "3"),
```
Comment (edwardhartnett) [inline spack/package.py]: This is surely not correct, there's no longer an ip2...

Comment (AlysonStahl-NOAA) [reply spack/package.py]: Should I just delete all references to ipolates = 3 and change it so we're only checking a boolean value? If so, would this cause issues if we decided we wanted to have checks where we build older versions of wgrib2?

Comment (edwardhartnett) [reply spack/package.py]: We will not be testing any older versions of wgrib2. Is there any other reason not to change it?

Comment (AlysonStahl-NOAA) [reply spack/package.py]: Nope. I'll go ahead and change it.

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**